impl/src/main/java/org/jboss/weld/util/Proxies.java

```diff
@@ -51,19 +54,28 @@
 
     public static class TypeInfo {
 
-        private final Set<Class<?>> interfaces;
-        private final Set<Class<?>> classes;
+        private final List<Class<?>> interfaces;
+        private final List<Class<?>> classes;
+        private final Comparator<Class<?>> classComparator = Comparator.comparing(Class::getName);
+        private boolean interfacesSorted;
```

Comment:
I don't think it makes sense to keep this class mutable and to introduce these flags. `TypeInfo.add(Type)` is only called in the `Proxies.TypeInfo.of(Set<? extends Type>)` method. So it would be probably reasonable to make this class immutable, i.e. create the sorted and unmodifiable lists in the constructor.

Reply:
This should now be addressed.